### DISCUSSION TITLE
Asiakkaat, joiden lainat on tehty kirjastossa jonka branchcode puuttuu kannasta.

### DISCUSSION THREAD

**Discussion by emta001** [KohaSuomi/Koha#1033] [Tarkistuspyynnöt]
Lumpeessa tuli vastaan tilanne, jossa asiakkaan Maksut-välilehden aukaiseminen johtaa 500 virheeseen. Tämä johtui siitä, että asiakkaan laina/lainat on tehty kirjastossa, jonka branchcode ei enää löydy tietokannasta (esim. Lumpeessa MLI_RA). Tarkistaisitteko omista kimpoistanne, löytyykö teiltä vastaavanlaisia lainoja? Ne saa kiinni seuraavalla rimpsulla:

```
SELECT iss.itemnumber, iss.branchcode AS "Lainakirjasto", i.homebranch AS "Niteen kotikirjasto" 
FROM issues iss 
LEFT JOIN items i ON(iss.itemnumber = i.itemnumber) 
WHERE iss.branchcode NOT IN( SELECT branchcode FROM branches );
```

**Comment by pknuuti** [comment]
Vaarassa on tällaisia lainoja 22 kpl. 
Niteet ovat kadonneita tai laskutettuja, joita ei saa poistaa tietokannasta. Niteiden kotikirjasto on vaihdettu, jolloin kirjaston on voinut poistaa tietokannasta.
![2023kuvat 276](https://github.com/KohaSuomi/Koha/assets/73887264/779597a0-ca3a-4303-95e6-c0576bdebc70)

**Comment by PiaKusmin** [comment]
Lapissa 11 kpl. Myös tosi vanhoja lainoja (laskutuksessa). Mitä näille voi tehdä tai voiko tehdä mitään?

**Reply by emta001** [reply]
Ne branchcodet pitää käydä päivittämässä olemassa olevaksi kirjastolyhenteeksi. Varmaan oikeimmin menee, jos tilalle päivittää niteen nykyisen kotikirjaston. Muuten noita on aika mahdoton arvata mitä niihin kuuluisi päivittää, eikä joitain pystykään jos kirjasto on jo poistettu.

**Reply by PiaKusmin** [reply]
Meillä kaikki on konversioaikaisia ja niissä kaikissa kirjastotiedon voi päivittää kotikirjastoksi. Teetkö sie muutoksen?

![image](https://github.com/KohaSuomi/Koha/assets/99241432/a5b212aa-fbe9-4cc7-b9f1-9c41218eb781)

**Comment by pknuuti** [comment]
> Ne branchcodet pitää käydä päivittämässä olemassa olevaksi kirjastolyhenteeksi. Varmaan oikeimmin menee, jos tilalle päivittää niteen nykyisen kotikirjaston. Muuten noita on aika mahdoton arvata mitä niihin kuuluisi päivittää, eikä joitain pystykään jos kirjasto on jo poistettu.

Miten se tapahtuu? Signumista muutetaan kotikirjaston lyhenne? Esim. tämä https://vaara.koha-suomi.fi/cgi-bin/koha/cataloguing/additem.pl?op=edititem&biblionumber=354312&itemnumber=10087669#edititem on laskutettu eikä sitä saa poistaa niin kauan kuin laskua ei ole hoidettu tavalla tai toisella (joko asiakas maksaa, mikä on epätodennäköistä tai lasku vanhenee ja se poistetaan saatavista)

**Reply by emta001** [reply]
Aattelin, että sen kotikirjaston voisi hakea ihan items.homebranch-sarakkeesta.

**Comment by MikkoLiimatainen** [comment]
Vaskista ei löytynyt raportilla yhtään tulosta.

**Comment by Tuomas-Kyyti** [comment]
Kyytistä löytyi 29 ja saa käydä muuttamassa niteen kotikirjaston lainakirjastoksi.

**Comment by pirliis** [comment]
OUTIsta ei löytynyt raportilla yhtään lainaa.

**Comment by pknuuti** [comment]
> Aattelin, että sen kotikirjaston voisi hakea ihan items.homebranch-sarakkeesta.

Eli käytännössä teetkö sinä/muu kehittäjä tämän muutoksen?

**Comment by emta001** [comment]
> > Aattelin, että sen kotikirjaston voisi hakea ihan items.homebranch-sarakkeesta.
> 
> Eli käytännössä teetkö sinä/muu kehittäjä tämän muutoksen?

Juu mie käyn tekemässä.

**Comment by emta001** [comment]
Lumme, Vaara, Lappi ja Kyyti korjattu seuraavasti:
`
UPDATE issues iss LEFT JOIN items i ON(iss.itemnumber = i.itemnumber) SET iss.branchcode = i.homebranch WHERE iss.branchcode NOT IN( SELECT branchcode FROM branches );`

**Comment by ReettaSiili** [comment]
Siilistä ei löytynyt mitään.

**Comment by ElinaUotila** [comment]
Kirkesistä  ei löytynyt näitä.

**Comment by Kati-Helle** [comment]
Hellestä raportti ei löytänyt yhtäkään tapausta.

**Comment by emta001** [comment]
Kiitos tarkistuksista! Ongelma on korjattu nyt kaikista kimpoista, joissa se esiintyi.
